rust/processor/src/db/common/models/fungible_asset_models/parquet_v2_fungible_asset_balances.rs

```diff
@@ -0,0 +1,245 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+// This is required because a diesel macro makes clippy sad
+#![allow(clippy::extra_unused_lifetimes)]
+#![allow(clippy::unused_unit)]
+
+use crate::{
+    bq_analytics::generic_parquet_processor::{GetTimeStamp, HasVersion, NamedTable},
+    db::common::models::{
+        coin_models::coin_utils::{CoinInfoType, CoinResource},
+        fungible_asset_models::{
+            v2_fungible_asset_activities::EventToCoinType,
+            v2_fungible_asset_balances::{
+                get_primary_fungible_store_address, CurrentFungibleAssetBalance,
+            },
+            v2_fungible_asset_utils::FungibleAssetStore,
+        },
+        object_models::v2_object_utils::ObjectAggregatedDataMapping,
+        token_v2_models::v2_token_utils::TokenStandard,
+    },
+    utils::util::standardize_address,
+};
+use ahash::AHashMap;
+use allocative_derive::Allocative;
+use aptos_protos::transaction::v1::{DeleteResource, WriteResource};
+use bigdecimal::{BigDecimal, Zero};
+use field_count::FieldCount;
+use parquet_derive::ParquetRecordWriter;
+use serde::{Deserialize, Serialize};
+
+#[derive(
+    Allocative, Clone, Debug, Default, Deserialize, FieldCount, ParquetRecordWriter, Serialize,
+)]
+pub struct FungibleAssetBalance {
+    pub txn_version: i64,
+    pub write_set_change_index: i64,
+    pub storage_id: String,
+    pub owner_address: String,
+    pub asset_type: String,
+    pub is_primary: bool,
+    pub is_frozen: bool,
+    pub amount: String, // it is a string representation of the u128
+    #[allocative(skip)]
+    pub block_timestamp: chrono::NaiveDateTime,
+    pub token_standard: String,
+}
+
+impl NamedTable for FungibleAssetBalance {
+    const TABLE_NAME: &'static str = "fungible_asset_balances";
+}
+
+impl HasVersion for FungibleAssetBalance {
+    fn version(&self) -> i64 {
+        self.txn_version
+    }
+}
+
+impl GetTimeStamp for FungibleAssetBalance {
+    fn get_timestamp(&self) -> chrono::NaiveDateTime {
+        self.block_timestamp
+    }
+}
+
+impl FungibleAssetBalance {
```

Comment:
do i understand correctly that this code is duplicated but with a different output(FungibleAssetBalance in Parquet vs FungibleAssetBalance in disiel)? can we abstract the transformation?